profiles/__init__.py

```diff
@@ -192,9 +193,36 @@ def handle_internal_error(e):
     raise e.original_exception
 
 
-@app.route("/health")
-def health():
-    """
-    Shows an ok status if the application is up and running
-    """
-    return {"status": "ok"}
+@app.route('/api/v1/healthcheck', methods=["GET"])
+def health_check():
+    #   Return codes
+    #       200 - Success
+    #       512 - LDAP failure
+    #       513 - DB failure
+    #       514 - LDAP & DB failure
+    return_code = 200
+    jsonout = {
+        "ldap": {
+            "status": True,
+            "server": None,
+        },
+        "database": {
+            "status": True,
+            "server": None,
+        }
+    }
+    try:
+        cshldap = csh_ldap.CSHLDAP(app.config['LDAP_BIND_DN'], app.config['LDAP_BIND_PASS'])
+        jsonout["ldap"]["server"] = cshldap.server_uri
+    except ldap.LDAPError:
+        jsonout["ldap"]["status"] = False
+        return_code = 512
+    try:
+        test_db = SQLAlchemy(app)
+        test_db.engine.connect()
```

Comment:
Similar, is there a way to check this without reinitialising every time?

Reply:
Couldn't find a straightforward way to do this for the db. There may exist a solution but I don't see this as being a major issue.

Reply:
Alright, LGTM